pints/_log_likelihoods.py

```diff
@@ -861,6 +862,302 @@ def evaluateS1(self, x):
+
+        # Convert the lower and upper limits to the correct type
+        a = self._convert_type(lower, limit_type="lower")
+        b = self._convert_type(upper, limit_type="upper")
+
+        if len(a) != self._no:
+            raise ValueError(
+                'Lower limit must be a ' +
+                ' scalar or a vector of length n_outputs.')
+
+        if len(b) != self._no:
+            raise ValueError(
+                'Upper limit must be a ' +
+                ' scalar or a vector of length n_outputs.')
+
+        diff = b - a
+        if np.any(diff <= 0):
+            raise ValueError('Upper limit ' +
+                             'must exceed lower limit.')
+        self._a = a
+        self._b = b
+
+        # Define the condition for whether a point is not censored
+        self._condition = (self._a < self._values
+                           ) & (self._values < self._b)
+
+        # Number of points that aren't censored (for each observation
+        # in the multioutput case)
+        self._n_not_censored = np.sum(self._condition, axis=0)
+
+    def _convert_type(self, limit, limit_type="lower"):
+
+        # If no lower or upper limit is supplied set it equal to
+        # -/+ infinity
+        if limit is None:
+            if limit_type == "lower":
+                limit = - np.inf
+            elif limit_type == "upper":
+                limit = np.inf
+
+        # Convert the limit to an object of the correct type
+        if limit is not None:
+            if np.isscalar(limit):
+                limit = np.ones(self._no) * float(limit)
+            else:
+                limit = pints.vector(limit)
+
+        return limit
+
+    def print_censored_values(self):
+
+        # Print data and values that are censored
+        print("The data are {}. \n The lower censored values"
+              " are {}. \n The upper censored values"
+              " are {}.".format(self._values,
+                                np.extract(self._values <= self._a,
+                                           self._values),
+                                np.extract(self._values >= self._b,
+                                           self._values)))
+
+    def __call__(self, x):
+        theta = np.asarray(x[:-self._no])
+        sigma = np.asarray(x[-self._no:])
+        if any(sigma <= 0):
+            return -np.inf
+
+        # Evaluate the problem output - do this only once as this is usually
+        # the most expensive step in inference, especially for ODE models
+        output = self._problem.evaluate(theta)
+
+        squared_error = np.sum((self._values - output)**2,
+                               axis=0, where=self._condition)
+
+        # Calculate part of the likelihood corresponding to the censored data
+        lower_censored_sum = np.sum(np.log(
+            scipy.stats.norm.cdf(x=self._a, loc=output, scale=sigma)),
+            where=self._values <= self._a)
```

Comment:
Cosmetic / performance comment:

I just realised that this mask also does never change, once initiated. Is there a reason why you only predefined self._condition but not the masks for lower and upper? I would suggest to define all masks in the same place: either everything in the __init__; or everything in the __call__. 

For large data, this may also be a cost worth front loading, which would, perhaps, favour the __init__ option.